google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkService.java

```diff
@@ -40,7 +44,46 @@ public static CloudSdkService getInstance() {
 
   public abstract void setSdkHomePath(String path);
 
-  public abstract Set<CloudSdkValidationResult> validateCloudSdk(@NotNull Path pathToCloudSdk);
+  protected abstract Set<CloudSdkValidationResult> validateCloudSdk(Path path);
+
+  /**
+   * Checks if the stored path contains a valid Cloud SDK.
+   */
+  public Set<CloudSdkValidationResult> validateCloudSdk() {
+    return validateCloudSdk(getSdkHomePath());
+  }
+
+  /**
+   * Checks if a given path is malformed.
+   *
+   * <p>Windows' implementation of Paths doesn't handle well converting strings with certain special
+   * characters to paths. This method should be called before {@code Paths.get(path)}.
```

Comment:
I think this javadoc is a bit confusing - it implies that all this method does is validate that the path is not malformed, when in reality it does all validations on the cloud sdk at that location. 

It might be good to change it to just say something more similar to the other validateCloudSdk methods.